src/bootstrap/compile.rs

```diff
@@ -724,8 +724,10 @@ fn copy_codegen_backends_to_sysroot(builder: &Builder,
     //
     // Here we're looking for the output dylib of the `CodegenBackend` step and
     // we're copying that into the `codegen-backends` folder.
-    let libdir = builder.sysroot_libdir(target_compiler, target);
-    let dst = libdir.join("codegen-backends");
+    let dst = builder.sysroot_libdir(target_compiler, target)
+        .parent()
```

Comment:
I believe we probably want to make this a dedicated function on builder. It seems like it's a relatively common thing to want (or at least that multiple places want it).